pkg/build/gobuild.go

```diff
@@ -826,7 +830,9 @@ func updatePath(cf *v1.ConfigFile, appPath string) {
 // Build implements build.Interface
 func (g *gobuild) Build(ctx context.Context, s string) (Result, error) {
 	// Determine the appropriate base image for this import path.
-	baseRef, base, err := g.getBase(ctx, s)
+	// We use the overall gobuild.ctx because the Build ctx gets cancelled
```

Comment:
Why does the Build ctx get cancelled early?

Reply:
https://github.com/google/ko/blob/a56047a2a1d500a646393b4a62d1d509bd5575e4/pkg/commands/resolver.go#L318

https://pkg.go.dev/golang.org/x/sync/errgroup#WithContext

We want to cancel outstanding builds if we encounter an error, so we use `errgroup.WithContext`, but the ctx is also cancelled once this returns: https://github.com/google/ko/blob/a56047a2a1d500a646393b4a62d1d509bd5575e4/pkg/commands/resolver.go#L415